lib/Http/RequestHelper.php

```diff
@@ -180,10 +180,11 @@ private function anonymizeIp(string $ip): string
      *
      * @internal
      */
-    public function createRequestWithContext(string $uri = '/'): Request
+    public function createRequestWithContext(string $uri = '/', string $host = 'localhost'): Request
     {
         $port = '';
         $scheme = $this->requestContext->getScheme();
+        $this->requestContext->setHost($host);
```

Comment:
Is it really required as the `Symfony\Component\Routing\RequestContext` is initialized with `localhost`as default value for host.

Reply:
I would go for a nullable `$host` property here and only set when the host is provided. wdyt?